### ISSUE TITLE
PrismLauncher hangs while installing a modpack

### ISSUE THREAD

**Issue by donottellmetonottellyou** [PrismLauncher/PrismLauncher#1056]
### Operating System

Linux

### Version of Prism Launcher

Prism Launcher 6.3

### Version of Qt

Qt 5.15.9

### Description of bug

When I tried to install Terrafirmacraft Reloaded 2.0.6 from the zip file, it hung (no network activity for 5-10 minutes) after downloading most of the modpack. I tried this three times.
![nonetwork](https://user-images.githubusercontent.com/115233539/236339213-cc7a508a-9340-453c-980b-ea77ecc103ba.png)

### Steps to reproduce

Install TerraFirmaCraft Reloaded on a high latency (ping 500-2000ms satellite) connection using the downloaded zip.

### Suspected cause

I think it may be caused by my high latency ping. A packet (or many) may drop, and Prism Launcher just waits forever for a download that will never complete. When I cancel the task, it shows it had a pending download:
```text
  2431.080 D | Download succeeded: "https://mediafilez.forgecdn.net/files/4484/171/TFCFlorae-1.18.2-2.0.5.jar"
  2814.009 D | Task "InstanceImportTask(0x55e38b43ab10)" aborted.
  2814.009 D | Task "InstanceStaging(0x55e38b8daf40)" aborted.
  2814.018 C | Aborted  "https://mediafilez.forgecdn.net/files/4484/719/tfc-hallucinocraft-1.18.2-0.0.16.jar"
  2814.018 D | Download aborted in previous step: "https://mediafilez.forgecdn.net/files/4484/719/tfc-hallucinocraft-1.18.2-0.0.16.jar"
  2814.019 D | Task "NetJob(Mod download)" aborted.
  2814.019 D | Task "FlameCreationTask(0x55e38b23ff10)" aborted.
  2814.019 C | Task "InstanceImportTask(0x55e38b43ab10)" aborted while not running!!!!
  2814.019 C | Task "InstanceStaging(0x55e38b8daf40)" aborted while not running!!!!
  2814.019 C | Task "FlameCreationTask(0x55e38b23ff10)" succeeded while not running!!!!
  2814.026 C | Task "InstanceImportTask(0x55e38b43ab10)" aborted while not running!!!!
```

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by donottellmetonottellyou** [comment]
Maybe a fix would be to have the multithreaded download be toggleable, because if the internet connection is saturated it makes it more likely to drop packets.

**Comment by Ryex** [comment]
this *might* be fixed by a combination of #1007 and #961.
Otherwise this is just a limitation of QT network manager we have no way to detect an effectively paused download or a download that has missed packets.

my work at an aria2 integration my bring a more complete fix but that will need to wait.

**Comment by Trial97** [comment]
Does it still happen on 7.2?

**Comment by donottellmetonottellyou** [comment]
I think it is better, but not completely solved.

I just tested downloading a modpack, and as my internet was working well today (no clouds to block sat) I had no problem downloading a small pack (Chemical Exchange). However, I tried downloading another modpack, and it froze when I simulated an interrupted connection by turning my wifi off for 5 seconds. It doesn't even attempt to retry the download for at least 15 minutes (I have to get to bed so I wasn't going to wait to see if it hung for longer), so I'd say that the issue still exists for bad connections.

![image](https://github.com/PrismLauncher/PrismLauncher/assets/115233539/6ef7d4b4-9b52-47bb-826e-d9309bcfb5af)

**Comment by Trial97** [comment]
Hi, can you please try to reproduce this using the latest nightly builds: https://nightly.link/PrismLauncher/PrismLauncher/workflows/trigger_builds/develop

**Comment by donottellmetonottellyou** [comment]
Hey, wonderful news, it seems to work well. I tried interrupting the internet both on the initial modpack download and the subsequent mod downloads. It survived and successfully downloaded the modpack.

One caveat, though it isn't a big deal: The first time I interrupted the download (the initial modpack download), when it resumed, it didn't update the progress bar and instead showed 50% with no progress details. I thought for a minute that it failed, but it just jumped ahead and started downloading the mods.
